test/integration/kubelet/watch_manager_test.go

```diff
@@ -40,7 +40,10 @@ import (
 func TestWatchBasedManager(t *testing.T) {
 	testNamespace := "test-watch-based-manager"
 	server := kubeapiservertesting.StartTestServerOrDie(t, nil, framework.DefaultTestServerFlags(), framework.SharedEtcd())
-	defer server.TearDownFn()
+	t.Cleanup(server.TearDownFn)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
```

Comment:
`k8s.io/kubernetes/test/utils/ktesting.Init` can be used to create a context which uses `t` for contextual logging and automatically gets canceled when the test is done.

Reply:
I've never known this library. How useful !
Thanks.